slo_lambda/src/windowed_slo.py

```diff
@@ -76,164 +56,74 @@ class SingleMetric(MetricBase):
       }
     """
 
-    def __init__(self, namespace: str, metric_name: str, dimensions: List):
+    def __init__(
+        self,
+        window_days: int,
+        namespace: str,
+        metric_name: str,
+        dimensions: List,
+        statistic: Optional[str] = "Sum",
+        extended_statistic: Optional[str] = None,
+        multiplier: float = 1.0,
+    ):
         self.namespace = namespace
         self.metric_name = metric_name
         self.dimensions = dimensions
+        self.statistic = statistic
+        self.extended_statistic = extended_statistic
+        self.multiplier = float(multiplier)
```

Comment:
@akrito - To allow having metrics that are negated or even scaled relative to others, how about this "multiplier" arg?

Reply:
sgtm